### PR TITLE
Add integration tests

### PR DESCRIPTION
**Pull request by ericphanson**
I've been making some changes and was worried about introducing issues that cause failures to package CI. So I thought I'd add some tests. Here I've added Oceananigans, cc @giordano.

### CODE REVIEW

File: integration/Oceananigans/Project.toml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+[deps]
+ExplicitImports = "7d51a73a-1435-4ff3-83d9-f097790105c7"
+Oceananigans = "9e8cae18-63c1-5223-a75c-80ca9d6e9a09"
+Test = "8dfed614-e22c-5e08-85e1-65c5234f0b40"
+
+[compat]
+Oceananigans = "=0.103.1"
```
Comment (giordano) [inline integration/Oceananigans/Project.toml]: Good to strictly fix the version, because I'm slowly but steadily going through all the modules to fix everything 😁

**Review by giordano — COMMENTED** [review]
Yay!  My only concern for you is that this will add a bit of precompilation latency for all the tests, but at least it's a real world large-ish codebase with many interesting corner cases.

File: .github/workflows/integration.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        os:
+          - ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      # Using `juliaup` over `setup-julia` so we can access beta versions easier
+      - uses: julia-actions/install-juliaup@v1
+        with:
+          julia-version: ${{ matrix.version }}
+      - uses: julia-actions/cache@v1
+      - uses: julia-actions/julia-buildpkg@v1
```
Comment (giordano) [inline .github/workflows/integration.yml]: I don't think this is really needed nowadays
```suggestion
```

File: .github/workflows/integration.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+          - Oceananigans
+    steps:
+      - uses: actions/checkout@v6
+      # Using `juliaup` over `setup-julia` so we can access beta versions easier
+      - uses: julia-actions/install-juliaup@v1
+        with:
+          julia-version: ${{ matrix.version }}
+      - uses: julia-actions/cache@v2
+      - name: Integration tests
+        run: julia --project=integration integration/runtest.jl ${{ matrix.project }}
```
Comment (ericphanson) [inline .github/workflows/integration.yml]:
```suggestion
        run: julia --color=yes --project=integration integration/runtest.jl ${{ matrix.project }}
```
